### PR TITLE
Use Kata-CC container host building meta-package for image build

### PR DESCRIPTION
**Pull request by miz060**
**What type of PR is this?**

**What this PR does / why we need it**:
Use Kata-CC container host building meta-package for Kata/CC image building.

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [x] tested upgrade from previous version

**Special notes for your reviewer**:
Validation
[TEST All VHDs] AKS Linux VHD Build - Msft Tenant:
https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=85813268&view=results

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh:30]: assuming the answer is no, but will we need to be sure to wait for your PR to update the UVM signing build to merge and rollout before merging this PR in? 

we've ran into issues in the past where new artifacts published by the signing build weren't picked up in time for official releases which ended up causing problems

Comment (miz060) [reply parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh:30]: Good point. No, we don't need to wait for updates on the UVM build and sign pipelines. They will produce the same artifacts before and after the update.

All the dependencies (kata-packages-host here) are on Azure Linux PMC so we are good to go.

**Review by cameronmeissner — APPROVED** [review]

**Review by anujmaheshwari1 — APPROVED** [review]

**Pull request merged**
